sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -7,46 +7,39 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
-using Azure.Core;
-using Azure.Core.Pipeline;
 using Azure.Messaging.ServiceBus;
-using Microsoft.Azure.WebJobs.Extensions.Clients.Shared;
 using Microsoft.Azure.WebJobs.Extensions.ServiceBus.Config;
-using Microsoft.Azure.WebJobs.Host;
 using Microsoft.Azure.WebJobs.Host.Executors;
 using Microsoft.Azure.WebJobs.Host.Listeners;
 using Microsoft.Azure.WebJobs.Host.Scale;
-using Microsoft.Extensions.Azure;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Listeners
 {
     internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider
     {
-        private readonly MessagingProvider _messagingProvider;
         private readonly ITriggeredFunctionExecutor _triggerExecutor;
-        private readonly string _functionId;
-        private readonly ServiceBusEntityType _entityType;
         private readonly string _entityPath;
         private readonly bool _isSessionsEnabled;
         private readonly bool _autoCompleteMessagesOptionEvaluatedValue;
         private readonly CancellationTokenSource _cancellationTokenSource;
         private readonly ServiceBusOptions _serviceBusOptions;
-        private readonly ILoggerFactory _loggerFactory;
         private readonly bool _singleDispatch;
         private readonly ILogger<ServiceBusListener> _logger;
 
         private readonly Lazy<MessageProcessor> _messageProcessor;
-        private Lazy<ServiceBusReceiver> _batchReceiver;
-        private Lazy<ServiceBusClient> _client;
-        private Lazy<SessionMessageProcessor> _sessionMessageProcessor;
-        private Lazy<ServiceBusScaleMonitor> _scaleMonitor;
+        private readonly Lazy<ServiceBusReceiver> _batchReceiver;
+        private readonly Lazy<ServiceBusClient> _client;
+        private readonly Lazy<SessionMessageProcessor> _sessionMessageProcessor;
+        private readonly Lazy<ServiceBusScaleMonitor> _scaleMonitor;
 
         private bool _disposed;
```

Comment:
Should we consider marking `_disposed`  as  `volatile`?

Reply:
yes I think it should be.